plugins/ptp/libptp/ptp.c

```diff
@@ -277,6 +277,8 @@ ptp_init_send_memory_handler(PTPDataHandler *handler,
 // free private struct + data
 static uint16_t
 ptp_exit_send_memory_handler (PTPDataHandler *handler) {
+	if (!handler->priv)
+		return PTP_RC_OK;
```

Comment:
Undo the above change.

This is a tricky one, as the execution path consists of 15 jumps. Anyway, trying to fix the bug at the end of the path won't work here. The best place to tackle the problem is much earlier, before Step 4 on the path:
Move Line 268 to right before Line 266.